spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/ReactiveResilience4JCircuitBreaker.java

```diff
@@ -32,28 +37,40 @@
 	private String id;
 	private Resilience4JConfigBuilder.Resilience4JCircuitBreakerConfiguration config;
 	private CircuitBreakerRegistry registry;
+	private List<Customizer<CircuitBreaker>> circuitBreakerCustomizers;
 
-	public ReactiveResilience4JCircuitBreaker(String id, Resilience4JConfigBuilder.Resilience4JCircuitBreakerConfiguration config, CircuitBreakerRegistry circuitBreakerRegistry) {
+	public ReactiveResilience4JCircuitBreaker(String id, Resilience4JConfigBuilder.Resilience4JCircuitBreakerConfiguration config,
+											  CircuitBreakerRegistry circuitBreakerRegistry,
+											  List<Customizer<CircuitBreaker>> circuitBreakerCustomizers) {
 		this.id = id;
 		this.config = config;
 		this.registry = circuitBreakerRegistry;
+		this.circuitBreakerCustomizers = circuitBreakerCustomizers;
 	}
 
 	@Override
 	public <T> Mono<T> run(Mono<T> toRun, Function<Throwable, Mono<T>> fallback) {
 		io.github.resilience4j.circuitbreaker.CircuitBreaker defaultCircuitBreaker = registry.circuitBreaker(id, config.getCircuitBreakerConfig());
-		Mono<T> toReturn = toRun.transform(CircuitBreakerOperator.of(defaultCircuitBreaker)).timeout(config.getTimeLimiterConfig().getTimeoutDuration());
+		circuitBreakerCustomizers.forEach(circuitBreakerCustomizer -> circuitBreakerCustomizer.customize(defaultCircuitBreaker));
```

Comment:
Is this because it's creating a new circuit breaker? I see customizers run also in `ReactiveResilience4JAutoConfiguration`?

Reply:
@spencergibb I generalized the `Customizer` interface so we can reuse it for more than one class.  That was why I made this change to the interface https://github.com/spring-cloud-incubator/spring-cloud-circuitbreaker/pull/11/files#diff-12830cd3449388dbbd2933706d312b0dL23
Other wise I would just be defining another `Customize` interface that could be used just for `CircuitBreakers` and it seemed kind of pointless to do that.

So right now we have two classes that we customize, `ReactiveResilience4JCircuitBreakerFactory` and `CircuitBreaker`.  The ``ReactiveResilience4JCircuitBreakerFactory` is customized after the auto config is created.  However we can't customize the `CircuitBreaker` till it is created later on, so the `CircuitBreaker` customizers are just passed along to the factory where we create them.

Reply:
@spencergibb does this make things clearer?